kernel-spark/src/main/java/io/delta/kernel/spark/read/SparkScan.java

```diff
@@ -258,4 +264,48 @@ CaseInsensitiveStringMap getOptions() {
   Configuration getConfiguration() {
     return hadoopConf;
   }
+
+  @Override
+  public NamedReference[] filterAttributes() {
+    return Arrays.stream(partitionSchema.fields())
+        .map(field -> FieldReference.column(field.name()))
+        .toArray(NamedReference[]::new);
+  }
+
+  @Override
+  public void filter(org.apache.spark.sql.connector.expressions.filter.Predicate[] predicates) {
+
+    // Try to convert runtime predicates to catalyst expressions, then create predicate evaluators
+    List<BasePredicate> evaluators = new ArrayList<>();
+    for (org.apache.spark.sql.connector.expressions.filter.Predicate predicate : predicates) {
+      // only the predicates on partition columns will be converted
+      Optional<Expression> catalystExpr =
+          dsv2PredicateToCatalystExpression(predicate, partitionSchema);
+      if (catalystExpr.isPresent()) {
+        BasePredicate predicateEvaluator =
+            org.apache.spark.sql.catalyst.expressions.Predicate.create(catalystExpr.get());
```

Comment:
Since this always happens on driver, shall we call createInterpreted to avoid codegen?